documentation/modules/ROOT/pages/transformations/event-changes.adoc

```diff
@@ -0,0 +1,147 @@
+:page-aliases: configuration/event-changes.adoc
+[id="event-record-changes"]
+= Event Record Changes
+
+:toc:
+:toc-placement: macro
+:linkattrs:
+:icons: font
+:source-highligher: highlight.js
+
+toc::[]
+
+ifdef::community[]
+[NOTE]
+====
+This single message transformation (SMT) is supported for only the SQL database connectors.
+====
+endif::community[]
+
+A {prodname} data change event has a complex structure that provides a wealth of information.
+However, in some cases, before a downstream consumer can process {prodname} change event messages, it requires additional information about field-level changes that result from the original database change.
+To enhance event messages with details about how a database operation modifies fields in the source database, {prodname} provides the `ExtractChangedRecordState` single message transformation (SMT).
+
+The event changes transformation is a
+link:https://kafka.apache.org/documentation/#connect_transforms[Kafka Connect SMT].
+
+== Change event structure
+
+
+{prodname} generates data change events that have a complex structure.
+Each event consists of three parts:
+
+* Metadata, which includes but is not limited to:
+
+** The type of operation that changed the data.
+** Source information, such as the names of the database and the table in which the change occurred.
+** Timestamp that identifies when the change was made.
+** Optional transaction information.
+
+* Row data before the change
+* Row data after the change
+
+The following example shows part of the structure of a typical {prodname} `UPDATE` change event:
+
+[source,json,indent=0]
+----
+{
+	"op": "u",
+	"source": {
+		...
+	},
+	"ts_ms" : "...",
+	"before" : {
+		"field1" : "oldvalue1",
+		"field2" : "oldvalue2"
+	},
+	"after" : {
+		"field1" : "newvalue1",
+		"field2" : "newvalue2"
+	}
+}
+----
+
+ifdef::community[]
+More details about change event structure are provided in
+xref:{link-connectors}[the documentation for each connector].
+endif::community[]
+
+The complex format of the message in the preceding example provides detailed information about changes that occur in the source database.
+However, the format might not be suitable for some downstream consumers.
+Sink connectors, or other parts of the Kafka ecosystem might expect the message to explicitly identify the fields that a database operation changes or leaves unchanged.
+The `ExtractChangedRecordState` SMT adds headers to the change event message to identify the fields that are modified by a database operation, and the fields that remain unchanged.
+
+[[event-changes-behavior]]
+== Behavior
+
+The event changes SMT extracts the `before` and `after` fields from a {prodname} `UPDATE` change event in a Kafka record.
+The SMT examines the fields in the `before` and `after` event state structures, and then writes the names of fields that changed or are unchanged based on the values of  settings that you apply to the SMT configuration.
```

Comment:
```suggestion
The transformation examines the `before` and `after` event state structures to identify the fields that are altered by an operation, and those that remain unchanged.
Depending on the connector configuration, the transformation then produces a modified event message that adds message headers to list the changed fields, the unchanged fields, or both.
```

Reply:
@Naros A few small wording changes to further unpack the meaning. If this looks ok to you, I'm good with releasing this to the wild.